linguatec_lexicon/importers.py

```diff
@@ -0,0 +1,457 @@
+                        "column": "{}!{}".format(sheet_name, e.code),
+                        "message": message,
+                    })
+                    continue
+
+                cleaned_data.append(word)
+        return cleaned_data
+
+    def write_to_database(cleaned_data):
+        count_entries = 0
+        for word in cleaned_data:
+            for entry in word.clean_entries:
+                entry.save()
+                entry.gramcats.set(entry.clean_gramcats)
+                count_entries += 1
+
+        sys.stdout.write("Imported: {} entries of {} words.".format(
+            count_entries, len(cleaned_data)))
+
+        return count_entries
+
+    def clean_variation():
+        if dry_run:
+            return None  # discard value because doesn't affect to data validation
+
+        # validate variation
+        try:
+            return DiatopicVariation.objects.get(name=variation)
+        except DiatopicVariation.DoesNotExist:
+            message = 'Diatopic variation "{}" does not exist'.format(variation)
+            save_ImportInfo_status(ii, ImportsInfo.FAILED, None, message)
+            raise CommandError(message)
+
+    # check that GramaticalCategories are initialized
+    if not GramaticalCategory.objects.all().exists():
+        message = ("There isn't any GramaticalCategory in the database. " +
+                   "Gramatical Categories should be initialized before importing " +
+                   "data for example running manage.py importgramcat.")
+        save_ImportInfo_status(ii, ImportsInfo.FAILED, None, message)
+        raise CommandError(message)
+
+    variation = clean_variation()
+
+    xlsx = pd.read_excel(input_file, sheet_name=None, header=None, usecols="A:C",
+                         names=['term', 'gramcats', 'translations'])
+    cleaned_data = populate_models(xlsx)
+
+    if errors:
+        save_ImportInfo_status(ii, ImportsInfo.FAILED, None, errors)
+
+    elif not dry_run:
+        try:
+            count_entries = write_to_database(cleaned_data)
+            save_ImportInfo_status(ii, ImportsInfo.COMPLETED, count_entries, None)
+        except IntegrityError as e:
+            save_ImportInfo_status(ii, ImportsInfo.FAILED, None, str(e))
+            raise CommandError("Error: " + e)
+
+    return errors, cleaned_data, xlsx
+
+
+def import_data(input_file, lexicon_pk, dry_run, allow_partial, imports_info_id=None):
+
+    errors = []
+    cleaned_data = {}
+    ii = get_ImportsInfo_objects(imports_info_id)
+
+    def read_input_file():
+        df = pd.DataFrame()
+
+        xlsx = pd.ExcelFile(input_file)
+
+        for sheet in xlsx.sheet_names:
+            # we define na_values and keep_default_na because defaults na_values
+            # includes empty string. We don't want that pandas replaces empty
+            # cells with 'nan'
+            partial = xlsx.parse(sheet, header=None, usecols='A:F')
+            # names=['colA', 'colB', 'colC', 'colD', 'colE', 'colF'])
```

Comment:
Remove unused code

Reply:
This was there before moving the code, I'm not sure if it is an unused line of code or just a comment